tests/tools/sinkhorn_divergence_test.py

```diff
@@ -70,15 +71,19 @@ def test_euclidean_point_cloud(self, cost_fn: costs.CostFn, epsilon: float):
     div2 -= 0.5 * sinkhorn.solve(geometry_xx, self._a, self._a).reg_ot_cost
     div2 -= 0.5 * sinkhorn.solve(geometry_yy, self._b, self._b).reg_ot_cost
 
-    np.testing.assert_allclose(div.divergence, div2, rtol=1e-5, atol=1e-5)
+    np.testing.assert_allclose(out.divergence, div2, rtol=1e-5, atol=1e-5)
 
-    # Test div of x to itself close to 0.
+    # Check differentiability of Sinkhorn divergence works, without NaN's.
+    grad = jax.grad(lambda x: div(x).divergence)(x)
+    assert jnp.all(jnp.logical_not(jnp.isnan(grad)))
```

Comment:
Use `np.testing...`.

Reply:
not sure I can find the adequate test in there, https://numpy.org/doc/stable/reference/routines.testing.html

Reply:
You can use `np.testing.assert_array_equal(jnp.isnan(grad), False)`.